### ISSUE TITLE
disconnect invalidates 1 active statement handle

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#279]
**Reported by efocht on 3 Nov 2006 16:37 UTC**
while testing branch-5-0 on fc5-x86_64 I get a flood of messages like
```
DBI::db=HASH(0x1563ba0)->disconnect invalidates 1 active statement handle
(either destroy statement handles or call finish on them before disconnecting)
at /opt/oscar-5.0a1r5399/lib/OSCAR/oda.pm line 171.
```
in the log.

**Comment by dikim33** [comment]
**Comment by efocht on 3 Nov 2006 17:20 UTC**
According to
http://www.unix.org.ua/orelly/perl/cookbook/ch14_11.htm
the statement handles need to be "finished". This doesn't seem to happen in ODA.
Where should it be done?

**Comment by dikim33** [comment]
**Comment by efocht on 3 Nov 2006 17:46 UTC**
Added some fixes to oda.pm. Needs testing...

**Comment by dikim33** [comment]
**Comment by efocht on 3 Nov 2006 18:45 UTC**
(In [5500]) Merged commit r5499 by efocht from 2006-11-03 18:46:28 +0100 (Fri, 03 Nov 2006)
with svnmerge_1by1 version v1.1 31.7.2006
from repository https://svn.oscar.openclustergroup.org/svn/oscar/trunk
to repository https://svn.oscar.openclustergroup.org/svn/oscar/branches/branch-5-0
Original commit message was:

Attempt to fix bug #279.
Added two $statement_handle->finish() calls.
Fixed weird indentation while trying to read the code...

Verified: works fine, no more messages.
Closes #279
